presto-main/src/main/java/com/facebook/presto/testing/TestingConnectorContext.java

```diff
@@ -137,6 +141,12 @@ public String formatRowExpression(ConnectorSession session, RowExpression expres
             {
                 return new RowExpressionFormatter(functionManager).formatRowExpression(session, expression);
             }
+
+            @Override
+            public Object optimizeRowExpression(ConnectorSession session, RowExpression expression, Function<VariableReferenceExpression, Object> variableResolver, ExpressionOptimizer.Level optimizationLevel)
+            {
+                return new RowExpressionInterpreter(expression, metadata, session, optimizationLevel).optimize(variableResolver::apply);
```

Comment:
I am wondering if we can use `ExpressionOptimizer` returned by `RowExpressionService` to do the optimization?

It looks to me the `RowExpressionOptimizer` did something very similar here (creating an `RowExpressionInterpreter`) and call `optimize`: https://github.com/prestodb/presto/blob/29c0ca12fb22f2c3bc9e35d7955c2060f541a3f3/presto-main/src/main/java/com/facebook/presto/sql/relational/RowExpressionOptimizer.java#L40

For this case, we need a variable resolver. I am wondering if adding another `optimize` method into `RowExpressionOptimizer` that takes customized variable resolver as well? cc @highker

Reply:
@wenleix @highker Good idea. Let me do that.